src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -493,6 +501,79 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  /**
+   * @param snapshotSummary
+   * @param passports RAS passports as JWT tokens
+   * @return ValidatePassportResult indicating whether the snapshot's contents are accessible via
+   *     one of the supplied RAS passports
+   */
+  public ValidatePassportResult verifyPassportAuth(
+      SnapshotSummaryModel snapshotSummary, List<String> passports) {
+    if (passports.isEmpty()) {
+      throw new InvalidAuthorizationMethod("No RAS Passports supplied for accessing snapshot");
+    }
+    if (!SnapshotSummary.passportAuthorizationAvailable(snapshotSummary)) {
+      throw new InvalidAuthorizationMethod("Snapshot cannot be accessed by RAS Passports");
+    }
+    String phsId = snapshotSummary.getPhsId();
+    String consentCode = snapshotSummary.getConsentCode();
+    var criterion = new RASv1Dot1VisaCriterion().consentCode(consentCode).phsId(phsId);
+    ecmService.addRasIssuerAndType(criterion);
+
+    var validatePassportRequest =
+        new ValidatePassportRequest().passports(passports).criteria(List.of(criterion));
+
+    return ecmService.validatePassport(validatePassportRequest);
+  }
+
+  @VisibleForTesting
+  public String passportInvalidForSnapshotErrorMsg(String userEmail) {
+    return String.format(
+        "Snapshot's passport criteria do not match %s's linked RAS passport", userEmail);
+  }
+
+  /**
+   * Throw if the user cannot access the snapshot via SAM permissions or linked RAS passports.
+   *
+   * @param snapshotId snapshot UUID
+   * @param userReq authenticated user
+   */
+  public void verifySnapshotAccessible(UUID snapshotId, AuthenticatedUserRequest userReq) {
+    boolean authorized = false;
+    List<String> causes = new ArrayList<>();
+    String userEmail = userReq.getEmail();
+    try {
+      iamService.verifyAuthorization(
+          userReq, IamResourceType.DATASNAPSHOT, snapshotId.toString(), IamAction.READ_DATA);
+      authorized = true;
+    } catch (Exception iamEx) {
+      logger.warn(
```

Comment:
If I'm reading this right, we're always going to try to verify with passports if Sam auth fails.  This is going to lead to confusing error messages I think (like if a snapshot isn't enabled for passport checking, I wouldn't expect to see exceptions about passports but I think that this would lead to that)